### DISCUSSION TITLE
Query returning a date field in the yyyy-mm-dd format.

### DISCUSSION THREAD

**Discussion by stayingcool** [hasura/graphql-engine#6278] [General]
I'm wondering if it is possible to have a query return a field in the yyyy-mm-dd format. 

**Example** (**orderDate** is a **timestamptz** field)
```
query GetOrdersByDate {
  orders(where: {orderDate: {_gte: "2020-11-29T00:00:00"}}) {
    id
    orderDate
    status
  }
}
```

**Result**
```
{
  "data": {
    "orders": [
      {
        "id": 2,
        "orderDate": "2020-11-29",
        "status": "Mailaway"
      }
    ]
  }
}
```

**Comment by praveenweb** [comment]
@stayingcool - You can create a postgres view for the `orders` table that parses the `orderDate` column as a date type instead of timestamp and you can query the view to get the response in the format you like.

**Reply by stayingcool** [reply]
@praveenweb - Thanks for the suggestion on Postgres view.

It'll be nice if there could be a way to transform the response data out of the box. Like some response mapping exposed where we could run a function or standard SQL functions to transform the response data before it is sent as API response.

**Note** - In AWS Appsync this can be done out of custom resolver response mapping (VTL). Which is essentially an ability to modify the SELECT (SQL DML) that pulls the response data and maps it to the API response.

**Reply by m-rgba** [reply]
> @praveenweb - Thanks for the suggestion on Postgres view.
> 
> It'll be nice if there could be a way to transform the response data out of the box. Like some response mapping exposed where we could run a function or standard SQL functions to transform the response data before it is sent as API response.
> 
> **Note** - In AWS Appsync this can be done out of custom resolver response mapping (VTL). Which is essentially an ability to modify the SELECT (SQL DML) that pulls the response data and maps it to the API response.

@stayingcool Something similar to Appsync's response mapping is also Hasura's computed fields: https://hasura.io/docs/1.0/graphql/core/schema/computed-fields.html - you could keep your table as is, but create a new "virtual" field for `formattedOrderDate`
